### DISCUSSION TITLE
Migrate from e-commerce

### DISCUSSION THREAD

**Discussion by Machaeon** [Blair2004/NexoPOS#1869] [Q&A]
Hi, I need to transfer products (thousands) from an online store database. I want to write a migration script and was wondering if such a request would be enough or could it create problems?

           INSERT INTO `ns_nexopos_products` (
                `name`, `tax_type`, `tax_group_id`, `tax_value`, `product_type`, `type`, `accurate_tracking`,
                `status`, `stock_management`, `barcode`, `barcode_type`, `sku`, `description`, `thumbnail_id`,
                `category_id`, `parent_id`, `unit_group`, `on_expiration`, `expires`, `searchable`, `author`, `uuid`,
                `created_at`, `updated_at`
            ) VALUES (
                '{product_name}', 'inclusive', NULL, 0.00000, 'product', 'materialized', 0,
                'available', 'enabled', '{product_barcode}', 'ean8', '{product_sku}', NULL, NULL,
                options['product']['category_id'], 0, options['product']['unit_group'], 'prevent-sales', 0, 1, options['product']['author'], NULL,
                '{current_datetime}', '{current_datetime}'
            );
            SET @last_product_id = LAST_INSERT_ID();
            INSERT INTO `ns_nexopos_products_unit_quantities` (
                `product_id`, `unit_id`, `barcode`, `quantity`, 
                `sale_price`, `sale_price_edit`, `sale_price_without_tax`, `sale_price_with_tax`, `sale_price_edit_without_tax`, 
                `wholesale_price`, `wholesale_price_edit`, `wholesale_price_without_tax`, `wholesale_price_with_tax`, `wholesale_price_edit_without_tax`
            ) VALUES (
                @last_product_id, {options['unit']['unit_id']}, '{product_barcode}-{options['unit']['unit_id']}', {product_quantity}, 
                {product_price}, {product_price}, {product_price}, {product_price}, {product_price}, 
                {product_price}, {product_price}, {product_price}, {product_price}, {product_price}
            );

**Comment by Blair2004** [comment]
Hi, 
importing products is not only about updating the "nexopos_products" table. In fact, there are 9 tables for the products (while not all tables need to be updated). Let me breakdown the use of each tables.

![image](https://github.com/Blair2004/NexoPOS/assets/5265663/93fc10a7-23c3-44bb-aa44-5d68c2efca34)

- nexopos_products: this keeps the products name, the category to which it's assigned, the tax group, the product type, SKU, status, unit_group, all information that are specific to a product basically.
- nexopos_products_galleries: that is used to store product images
- nexopos_products_categories: this is where product categories are stored.
- nexopos_products_histories: keep an history of all operations of each product (sales, returns, etc).
- nexopos_products_histories_combined: combine for a day the previous table (nexopos_products_histories)
- nexopos_products_metas: not really used, this table will be deprecated
- nexopos_products_subitems: for grouped products, this keeps a reference to the items included on the group
- nexopos_products_taxes: store the computation of taxes per products
- nexopos_products_unit_quantities: this table is important as it keeps a reference to the stock for each product unit. Note that on NexoPOS, a product can be sold as multiple units. This table keeps a reference to all those unit and the stock available for each unit.

So if you have to import products that way, I'll suggest you updating:
- nexopos_products
- nexopos_products_categories (products needs to be assigned to a category)
- nexopos_products_unit_quantities

**Reply by Machaeon** [reply]
Yes, I already created the category structure and now I just need to migrate the products (linking them to the categories, of course). So adding entries to nexopos_products and nexopos_products_unit_quantities is a sufficient minimum in my case?
